### ISSUE TITLE
Slow application start if recently used file is on unavailable network

### ISSUE THREAD

**Issue by jamesobutler** [Slicer/Slicer#6631]
As originally discussed at https://discourse.slicer.org/t/slow-application-start-if-recently-used-file-is-on-unavailable-network/25970/5.

While at work I opened a file in 3D Slicer that I had accessed from a local file share that is on our corporate network. When I went home and tried to start Slicer it was hanging at the "Initializing user interface..." step on the splash screen for 45 seconds before proceeding to load the Slicer modules. This was due to a setting for the recent files including the path to the local file share that I no longer had access to from my home network.

https://user-images.githubusercontent.com/15837524/198886694-549dc9ef-0bd9-4a79-81fd-a109248d2008.mp4

@lassoan has suggested maybe changing this to a runtime check rather than at startup. I also don't see a reason to wait 45 seconds trying to see if a file is still accessible from a file share location.

The specific code of interest is this section in `restoreGUIState`:
https://github.com/Slicer/Slicer/blob/9a56f644406357394f42004133c6ee13ae7e3ad3/Base/QTApp/qSlicerMainWindow.cxx#L1694-L1702

## Environment
- Slicer version: Slicer-5.1.0-2022-10-28, Slicer 5.0.3
- Operating system: Windows

**Comment by lassoan** [comment]
> don't see a reason to wait 45 seconds trying to see if a file is still accessible from a file share location

This timeout is an operating system setting. Usually the timeout is 30 seconds, which is reasonable because often a file server has to return from a low-power state, spin up disks, etc.

**Comment by jamesobutler** [comment]
I wonder if other applications do this type of I/O on external files at startup. I was just surprised Slicer would be attempting to read a file in a network location at startup.

Maybe there could be different startup rules if the recent files are specific at a network location. 

During my debugging I originally thought it was maybe an issue with my Slicer DICOM database being in a OneDrive location (my corporate laptop is setup by default to sync the user documents folder to OneDrive), but that wasn’t the issue.  I also thought maybe a slow to respond Slicer server providing version info at startup, but that also wasn’t the case here.

**Comment by lassoan** [comment]
Yes, I find this delay to be quite common, but not at application startup, but whenever you attempt to access a network resource. I remember happening even when I opened a system dialog for opening a file and had some network drive or favorite folder that was not accessible (I think the file dialog tried to determine what icon to display for it).
